src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -531,6 +534,17 @@ private void stopConnection(String errorMessage) {
             transportEnum = TransportEnum.ALL;
             this.localHeaders.clear();
             this.streamMap.clear();
+            if (this.pingTimer != null) {
+                this.pingTimer.cancel();
+            }
```

Comment:
The pingTimer is already being canceled a few lines above in this method. There was a community PR that added it recently.